form_api_backend.py

```
@@ -321,16 +378,20 @@ def download_form(form_id: str, file_name: str):
-            mimetype='application/pdf'
-        )
+        # Atomic check and serve - avoid race condition by using try/except
+        try:
+            return send_file(
+                file_path,
+                as_attachment=True,
+                download_name=file_name,
+                mimetype='application/pdf'
+            )
+        except FileNotFoundError:
+            raise NotFound("File not found")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)